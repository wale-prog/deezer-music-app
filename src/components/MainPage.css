.artist-tile {
  display: grid;
  grid-template-columns: 187px 187px;
  justify-content: center;
  border: none;
}

.artist-tile a {
  text-decoration: none;
  color: #d3d3d3;
}

.artist-tile a:nth-child(even) {
  background-color: rgb(65 103 176);
}

.artist-tile a:nth-child(odd) {
  background-color: rgb(63 98 164);
}

.main-page {
  background-color: rgb(63 98 164);
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 420px;
  margin: 0 auto;
}

.search-bar {
  margin: 20px 0;
  width: 375px;
  border: 2px solid #d3d3d3;
  height: 26px;
  border-radius: 6px;
  background-color: rgb(63 98 164);
}

::placeholder {
  color: #d3d3d3;
  margin-left: 12px;
}
