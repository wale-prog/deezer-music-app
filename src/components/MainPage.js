import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useSearchParams, Link } from 'react-router-dom';
import { getChart } from '../redux/ChartsSlice';
import ArtistTile from './ArtistTile';

const MainPage = () => {
  const charts = useSelector((state) => state.charts);
  const dispatch = useDispatch();

  useEffect(() => {
    if (charts.length === 0) {
      dispatch(getChart());
    }
  }, []);

  const [search, setSearch] = useSearchParams();

  return (
    <div className="main-page">
      <input
        className="search-bar"
        type="text"
        value={search.get('filter') || ''}
        placeholder="Search music by title..."
        onChange={(event) => {
          const filter = event.target.value;
          if (filter) {
            setSearch({ filter });
          } else {
            setSearch({});
          }
        }}
      />

      <div className="artist-tile">
        {charts
          .filter((item) => {
            const filter = search.get('filter');
            if (!filter) return true;
            const artistName = item.title.toLowerCase();
            return artistName.startsWith(filter.toLocaleLowerCase());
          })
          .map((item) => (
            <Link key={item.id} to={`/${item.artist.name}`}>
              <ArtistTile
                key={item.id}
                data={item}
              />
            </Link>
          ))}
      </div>
    </div>

  );
};
export default MainPage;
