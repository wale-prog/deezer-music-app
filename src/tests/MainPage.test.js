import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/index';
import MainPage from '../components/MainPage';

describe('Test the Main page renders as expected', () => {
  it('Main page is rendered', () => {
    const mainPage = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <MainPage />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(mainPage).toMatchSnapshot();
  });
});
