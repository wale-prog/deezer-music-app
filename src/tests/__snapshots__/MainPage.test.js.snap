// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Main page renders as expected Main page is rendered 1`] = `
<div
  className="main-page"
>
  <input
    className="search-bar"
    onChange={[Function]}
    placeholder="Search music by title..."
    type="text"
    value=""
  />
  <div
    className="artist-tile"
  />
</div>
`;
